client/blockchain-service/src/commands.rs

```diff
@@ -368,11 +373,15 @@ pub trait BlockchainServiceInterface {
         bucket_id: BucketId,
     ) -> Result<Option<MainStorageProviderId>, QueryMspIdOfBucketIdError>;
 
-    /// Helper function to release the forest root write lock.
+    /// Helper function to release the Forest root write lock.
     async fn release_forest_root_write_lock(
         &self,
         forest_root_write_tx: tokio::sync::oneshot::Sender<()>,
     ) -> Result<()>;
+
+    /// Get the current Forest root for a given Provider.
```

Comment:
This is tricky. If you plan to use this later on in the tasks instead of the hard-coded constant it will be wrong.
I think the better approach is to pass the forest key with the event itself to avoid having tasks spawned by "fork" A with the snapshot from fork B.